### PR TITLE
Outdoorsman Refactor: Camping

### PR DESCRIPTION
**Pull request by tf-4**
## About The Pull Request

Updates the sleeping bag code allowing for more enjoyable camping. Sure to keep you snuggly and warm.

I'll be doing the tents in my next PR.

Semi-WIP

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
Yeah I'm not putting a screenshot of that on GitHub

</details>

## Changelog
:cl:
refactor: Updated sleeping bag code. Report any weirdness.
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_clothing/kinky_sleepbag.dm [hunk 1]
```diff
@@ -1,196 +1,245 @@
...
+#define SLOWDOWN_INFLATED 14
+#define SLOWDOWN_DEFLATED 4
+#define FOLD_VOLUME 40
+#define PROCESS_NOISE_VOLUME 100
+#define ICON_STATE_FOLDED_PINK "sleepbag_pink_deflated_folded"
+#define ICON_STATE_FOLDED_TEAL "sleepbag_teal_deflated_folded"
+#define BAG_NOISE_CHANCE 1
+
+#define BREAKOUT_TIME_INFLATED 10 SECONDS
+#define BREAKOUT_TIME_DEFLATED 5 SECONDS
```
Comment (tf-4) [inline modular_skyrat/modules/modular_items/lewd_items/code/lewd_clothing/kinky_sleepbag.dm:12]: this is just for my sanity while testing, before someone complains

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_clothing/kinky_sleepbag.dm [hunk 2]
```diff
@@ -1,196 +1,245 @@
...
+		if(.)
+			return
+		var/choice = show_radial_menu(human_user, src, bag_colors, custom_check = CALLBACK(src, PROC_REF(check_menu), human_user), radius = 36, require_near = TRUE)
+		if(!choice)
+			return FALSE
+		bag_color = choice
+		update_icon()
+		// update_icon_state()
+		color_changed = TRUE
+		return
```
Comment (tf-4) [inline modular_skyrat/modules/modular_items/lewd_items/code/lewd_clothing/kinky_sleepbag.dm]: we really need a better solution to this, probably a reskin element or something

Comment (softcerv) [reply modular_skyrat/modules/modular_items/lewd_items/code/lewd_clothing/kinky_sleepbag.dm]: That sounds like a good idea for a PR

Comment (tf-4) [reply modular_skyrat/modules/modular_items/lewd_items/code/lewd_clothing/kinky_sleepbag.dm]: I'm vaguely working on it yeah. Not with any sort of haste so feel free if you fancy it.

**Pull request closed without merging**